### DISCUSSION TITLE
How to generate spatial and temporal baseline plots in ISCE (or other software)?

### DISCUSSION THREAD

**Discussion by leeyhgeo** [isce-framework/isce2#651] [Q&A] [answered]
Hello,

I'm planning to conduct SBAS InSAR analysis and I've collected all ERS, Envisat, and Sentinel-1 SAR images over my study area. Now I would like to draw **spatial and temporal baseline plots** (as the figure below) of these images to determine which image pairs are suitable for further analysis. However, I have no idea how to get baseline information. I'm wondering if there is any function in ISCE (or other software like SNAP) that can generate spatial and temporal baseline plots.

Thank you in advance!

![Baseline-plots-of-interferograms-used-in-SBAS-a-and-PSInSAR-b-The-interferograms](https://user-images.githubusercontent.com/105381356/214261317-54cee00a-2344-4d8d-aa83-1048e45332db.png)
Figure source: https://www.researchgate.net/figure/Baseline-plots-of-interferograms-used-in-SBAS-a-and-PSInSAR-b-The-interferograms_fig1_348937233

**Comment by fdelgadodelapuente** [comment, marked as answer]
The stripmap stack processor will generate such a plot for ERS and ENVISAT data.

**Reply by leeyhgeo** [reply]
That's great! Thanks for your information. I'm struggling to process ERS and Envisat data through stackStripMap.py.

**Reply by AP2607** [reply]
How to plot for S-1 IW?

**Reply by fdelgadodelapuente** [reply]
The TOPS stack processor does not have a routine to generate such a plot. You will have to extract the baseline information and create one in either Python or Matlab.

**Reply by AP2607** [reply]
Is it possible to extract baseline information about the entire dataset (without setting master acquisition)?

**Reply by fdelgadodelapuente** [reply]
No, you need to set a reference image. However, topsApp.py will calculate the Perp for every processed pair at step computeBaselines

You can also use the SBAS utility at ASF Vertex to generate a baseline plot

**Reply by AP2607** [reply]
Ok thanks.

**Comment by fdelgadodelapuente** [comment]
First, you need to unpack the data with unpackFrame_ENV.py and unpackFrame_ERS.py

**Reply by leeyhgeo** [reply]
I've tried unpackFrame_ERS.py and I encounter errors. I think there're some bugs in the original code. I'm trying to debug. Have you tried unpackFrame_ERS.py and faced some errors as well?

**Comment by fdelgadodelapuente** [comment]
Yes, it works fine. You need to edit the code to set tup the path to the ERS-1/2 ODR/PCR orbits.
